### PR TITLE
Bump tarantool to 2.10.4

### PR DESCRIPTION
**Pull request by foxzi**
* Bump tarantool version to 2.10.4
* Update gperftools version to 2.10

### CODE REVIEW

File: dockerfiles/alpine_3.15 [hunk 1]
```diff
@@ -14,7 +14,7 @@ ENV TARANTOOL_VERSION=${TNT_VER} \
     TARANTOOL_DOWNLOAD_URL=https://github.com/tarantool/tarantool.git \
     TARANTOOL_INSTALL_LUADIR=/usr/local/share/tarantool \
     GPERFTOOLS_REPO=https://github.com/gperftools/gperftools.git \
-    GPERFTOOLS_TAG=gperftools-2.9.1 \
+    GPERFTOOLS_TAG=gperftools-2.10 \
     LUAROCKS_URL=https://github.com/tarantool/luarocks/archive/6e6fe62d9409fe2103c0fd091cccb3da0451faf5.tar.gz \
```
Comment (NickVolynkin) [inline dockerfiles/alpine_3.15:18]: Why do we use a commit from year 2017? https://github.com/tarantool/luarocks/commit/6e6fe62d9409fe2103c0fd091cccb3da0451faf5

File: dockerfiles/alpine_3.16 [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+       mkdir -p ${GOPATH}/src ${GOPATH}/bin && \
+       go install github.com/google/pprof@latest ) \
+    && : "---------- tarantool ----------" \
+    && mkdir -p /usr/src/tarantool \
+    && git clone "$TARANTOOL_DOWNLOAD_URL" /usr/src/tarantool \
+    && git -C /usr/src/tarantool checkout "$TARANTOOL_VERSION" \
+    && git -C /usr/src/tarantool submodule update --init --recursive \
+    && (cd /usr/src/tarantool; \
+       echo "WARNING: Temporary fix for test/unit/cbus_hang test" ; \
+       git cherry-pick d7fa6d34ab4e0956fe8a80966ba628e0e3f81067 2>/dev/null || \
```
Comment (NickVolynkin) [inline dockerfiles/alpine_3.16:108]: Commit "Fix compilation for Alpine 3.5". Is it still relevant? Why do we make a cherry-pick of a commit which is already in our branch?
https://github.com/tarantool/tarantool/commit/d7fa6d34ab4e0956fe8a80966ba628e0e3f81067

File: versions/ubuntu_20.04_2.10.4 [hunk 3]
```diff
@@ -0,0 +1 @@
+2.10.4.g816000e10-1
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply versions/ubuntu_20.04_2.10.4:1]: `816000e10-1` is a part of the name of the corresponding Ubuntu package (`tarantool_2.10.4.g816000e10-1_arm64.deb`).

File: dockerfiles/alpine_3.15 (hunk 1 above)
Comment (ylobankov) [reply dockerfiles/alpine_3.15:18]: I guess it was moved without any changes from alpine_3.9, that's all.

**Review by ylobankov — APPROVED** [review]
LGTM

File: dockerfiles/alpine_3.16 [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+# An ARG instruction goes out of scope at the end of the build
+# stage where it was defined. To use an arg in multiple stages,
+# each stage must include the ARG instruction
+ARG TNT_VER
+ARG NPROC
+ENV TARANTOOL_VERSION=${TNT_VER} \
+    TARANTOOL_DOWNLOAD_URL=https://github.com/tarantool/tarantool.git \
+    TARANTOOL_INSTALL_LUADIR=/usr/local/share/tarantool \
+    GPERFTOOLS_REPO=https://github.com/gperftools/gperftools.git \
+    GPERFTOOLS_TAG=gperftools-2.9.1 \
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply dockerfiles/alpine_3.16]: This file was removed from the patch.

File: tests/building.sh [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+docker manifest rm tarantool/tarantool:2.10-nogc64-centos7
+docker manifest rm tarantool/tarantool:2.10.4-nogc64-centos7
+
+docker manifest create tarantool/tarantool:2-nogc64-centos7 tarantool/tarantool:2.10.4-nogc64-centos7-amd64
+docker manifest create tarantool/tarantool:2.10-nogc64-centos7 tarantool/tarantool:2.10.4-nogc64-centos7-amd64
+docker manifest create tarantool/tarantool:2.10.4-nogc64-centos7 tarantool/tarantool:2.10.4-nogc64-centos7-amd64
+
+docker manifest push tarantool/tarantool:2-nogc64-centos7
+docker manifest push tarantool/tarantool:2.10-nogc64-centos7
+docker manifest push tarantool/tarantool:2.10.4-nogc64-centos7
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply tests/building.sh]: This file was removed from the patch.

File: tests/building.sh [hunk 6]
```diff
@@ -0,0 +1,108 @@
+# x86
+time TAG=2.10.4-gc64-ubuntu-amd64 OS=ubuntu DIST=20.04 VER=2.10.4 NPROC=8 PORT=5200 GC64=ON make -f .gitlab.mk build
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply tests/building.sh]: This file was removed from the patch.

File: tests/building.sh [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+time TAG=2.10.4-nogc64-ubuntu-amd64 OS=ubuntu DIST=20.04 VER=2.10.4 NPROC=8 PORT=5200 GC64=OFF make -f .gitlab.mk build
+
+time TAG=2.10.4-gc64-centos7-amd64 OS=centos DIST=7 VER=2.10.4 NPROC=8 PORT=5200 GC64=ON make -f .gitlab.mk build
+time TAG=2.10.4-nogc64-centos7-amd64 OS=centos DIST=7 VER=2.10.4 NPROC=8 PORT=5200 GC64=OFF make -f .gitlab.mk build
+
+time TAG=2.10.4-nogc64-amd64 OS=alpine DIST=3.15 VER=2.10.4 NPROC=8 PORT=5200 LUAJIT_DISABLE_SYSPROF=ON GC64=OFF make -f .gitlab.mk build
+time TAG=2.10.4-gc64-amd64 OS=alpine DIST=3.15 VER=2.10.4 NPROC=8 PORT=5200 LUAJIT_DISABLE_SYSPROF=ON GC64=ON make -f .gitlab.mk build
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply tests/building.sh]: This file was removed from the patch.

File: dockerfiles/alpine_3.16 [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+    && git -C /usr/src/gperftools checkout "$GPERFTOOLS_TAG" \
+    && (cd /usr/src/gperftools; \
+        ./autogen.sh; \
+        ./configure --disable-libunwind --enable-frame-pointers; \
+        make -j ${NPROC}; \
+        cp .libs/libprofiler.so* /usr/local/lib/) \
+    && : "---------- pprof for gperftools ----------" \
+    && : "To avoid of the issue:" \
+    && : "'The master branch is Incompatible with go version < 1.13 #538'" \
+    && : "https://github.com/google/pprof/issues/538" \
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply dockerfiles/alpine_3.16]: This file was removed from the patch.

File: dockerfiles/alpine_3.16 (hunk 2 above)
Comment (ylobankov) [reply dockerfiles/alpine_3.16]: This file was removed from the patch.

File: versions/alpine_3.15_2.10.4 [hunk 9]
```diff
@@ -0,0 +1 @@
+2.10.4-1-g02908e625
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply versions/alpine_3.15_2.10.4]: I believe we can do it, but for now we build 2.10.4 on the base of alpine 3.15.

**Pull request merged**